contracts/src/token/erc721/extensions/metadata.rs

```diff
@@ -58,3 +61,24 @@ impl IErc721Metadata for Erc721Metadata {
         self._base_uri.get_string()
     }
 }
+
+impl IErc165 for Erc721Metadata {
+    fn supports_interface(interface_id: FixedBytes<4>) -> bool {
+        <Self as IErc721Metadata>::INTERFACE_ID
+            == u32::from_be_bytes(*interface_id)
+    }
+}
+
+#[cfg(all(test, feature = "std"))]
+mod tests {
+    // use crate::token::erc721::extensions::{Erc721Metadata, IErc721Metadata};
+
+    // TODO: IErc721Metadata should be refactored to have same api as solidity
```

Comment:
TODO

Reply:
You implemented metadata extension differentl from solidity. It has `base_uri` function instead of `token_uri`. And now it exposes "baseUri()" function into abi that is not like that in solidity. I think we will refactor it later